### ISSUE TITLE
Implement Auth.sendCustomChallengeAnswer() allowing multiple custom challenges for the CUSTOM_AUTH flow

### ISSUE THREAD

**Issue by ansonallard** [aws-amplify/amplify-swift#1444]
### Is your feature request related to a problem? Please describe.

I am attempting to implement password-less (one-time password) authentication for my iOS application using Cognito and Amplify. I have my authentication setup properly on the client side. I followed this blog post when setting up my cloud-side implementation: https://aws.amazon.com/blogs/mobile/implementing-passwordless-email-authentication-with-amazon-cognito/

The above blog post introduces a check that allows a user 3 attempts to input the correct password.

For my client, I make use of the sample code defined in the iOS documentation found here: https://docs.amplify.aws/lib/auth/signin_with_custom_flow/q/platform/ios/#sign-in-a-user

Client:

Sign In:
```
func signIn(username: String) {
    Amplify.Auth.signIn(username: username) { result in
        switch result {
        case .success:
            if case .confirmSignInWithCustomChallenge(_) = result.nextStep {
                // Ask the user to enter the custom challenge.
            } else {
                print("Sign in succeeded")   
            }
        case .failure(let error):
            print("Sign in failed \(error)")
        }
    }
}
```

After the signIn call and they receive their one-time password, I make the confirmSignIn call:
```
func customChallenge(response: String) {
    Amplify.Auth.confirmSignIn(challengeResponse: response) { result in
        switch result {
        case .success:
            print("Confirm sign in succeeded")   
        case .failure(let error):
            print("Confirm sign in failed \(error)")
        }
    }
}
```

If I get the code correct, I am logged in as expected. However, if I enter the code incorrectly, the callback from Amplify.Auth.confirmSignIn is never hit, and my app sits waiting. 

If I change my cloud-side code to support one log-in attempt (code found here: https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-define-auth-challenge.html#aws-lambda-triggers-define-auth-challenge-example), which is not a desirable behavior for the user, the `confirmSignIn` method returns as expected, either letting the user in if the code was correct, or denying entry and ending the authentication flow if the code was incorrect.

In the Javascript implementation of authentication they have two methods for "confirming a sign in request" (documentation here: https://docs.amplify.aws/lib/auth/advanced/q/platform/js/#lambda-triggers):

- `Auth.confirmSignIn`

- `Auth.sendCustomChallengeAnswer`

Since my cloud resources support multiple challenges, so the user has multiple attempts to enter the code, I would like to see the iOS SDK add the `Auth.sendCustomChallengeAnswer` function, thereby supporting more than one custom challenge. 
 

### Describe the solution you'd like

I'd like to see the `Auth.sendCustomChallengeAnswer` method implemented with an object that has a nextStep field, like the `Auth.signIn` method:

``` if case .confirmSignInWithCustomChallenge(_) = result.nextStep {```

This would allow for more than one challenge.

### Describe alternatives you've considered

None at this time.

### Is the feature request related to any of the existing Amplify categories?

Auth

### Additional context

The Javascript SDK has this method already implemented, but they had some issues with it that users discovered in 2019. I'm linking the issue for reference if it is helpful: https://github.com/aws-amplify/amplify-js/issues/3373

This function is not availabe in Android either: https://github.com/aws-amplify/aws-sdk-android/issues/2653

**Comment by ruisebas** [comment]
Thanks for opening this feature request. We'll evaluate it and respond to this issue with any updates.

**Comment by royjit** [comment]
Supporting for more than one custom challenge should be fixed  in [v2.0.0](https://github.com/aws-amplify/amplify-swift/releases/tag/v2.0.0)
